features/steps/steps.py

```diff
@@ -186,6 +202,18 @@ def step_impl(context, field, values):
 
     context.applicable = getattr(context, 'applicable', True) and applicable
 
+
+@given('The {entity} is a part of another {other_entity} through the relationship {relationship}')
+def step_impl(context, entity, other_entity, relationship):
+    relationships = context.model.by_type(relationship)
+    instances = []
+    for rel in relationships:
+        if rel.RelatingObject.is_a(other_entity):
```

Comment:
This end of the relationship is not always called `RelatingObject` see table below (only for IFC4).

A quick hack would be:

```
>>> import ifcopenshell
>>> f = ifcopenshell.file()
>>> rel = f.createIfcRelAggregates()
>>> [d for d in dir(rel) if d.startswith('Relating')][0]
'RelatingObject'
```

But we can also build a lookup table.

entity | attribute | cardinality | ordered | type
-- | -- | -- | -- | --
IfcRelAssigns | RelatedObjects | 1..* | n | IfcObjectDefinition
IfcRelAssignsToActor | RelatingActor | 1..1 |   | IfcActor
IfcRelAssignsToControl | RelatingControl | 1..1 |   | IfcControl
IfcRelAssignsToGroup | RelatingGroup | 1..1 |   | IfcGroup
IfcRelAssignsToProcess | RelatingProcess | 1..1 |   | IfcProcess,IfcTypeProcess
IfcRelAssignsToProduct | RelatingProduct | 1..1 |   | IfcProduct,IfcTypeProduct
IfcRelAssignsToResource | RelatingResource | 1..1 |   | IfcResource,IfcTypeResource
IfcRelAssociates | RelatedObjects | 1..* | n | IfcObjectDefinition,IfcPropertyDefinition
IfcRelAssociatesApproval | RelatingApproval | 1..1 |   | IfcApproval
IfcRelAssociatesClassification | RelatingClassification | 1..1 |   | IfcClassification,IfcClassificationReference
IfcRelAssociatesConstraint | RelatingConstraint | 1..1 |   | IfcConstraint
IfcRelAssociatesDocument | RelatingDocument | 1..1 |   | IfcDocumentInformation,IfcDocumentReference
IfcRelAssociatesLibrary | RelatingLibrary | 1..1 |   | IfcLibraryInformation,IfcLibraryReference
IfcRelAssociatesMaterial | RelatingMaterial | 1..1 |   | IfcMaterialDefinition,IfcMaterialList,IfcMaterialUsageDefinition
IfcRelConnectsElements | RelatingElement | 1..1 |   | IfcElement
IfcRelConnectsElements | RelatedElement | 1..1 |   | IfcElement
IfcRelConnectsPortToElement | RelatingPort | 1..1 |   | IfcPort
IfcRelConnectsPortToElement | RelatedElement | 1..1 |   | IfcDistributionElement
IfcRelConnectsPorts | RelatingPort | 1..1 |   | IfcPort
IfcRelConnectsPorts | RelatedPort | 1..1 |   | IfcPort
IfcRelConnectsStructuralActivity | RelatingElement | 1..1 |   | IfcElement,IfcStructuralItem
IfcRelConnectsStructuralActivity | RelatedStructuralActivity | 1..1 |   | IfcStructuralActivity
IfcRelConnectsStructuralMember | RelatingStructuralMember | 1..1 |   | IfcStructuralMember
IfcRelConnectsStructuralMember | RelatedStructuralConnection | 1..1 |   | IfcStructuralConnection
IfcRelContainedInSpatialStructure | RelatedElements | 1..* | n | IfcProduct
IfcRelContainedInSpatialStructure | RelatingStructure | 1..1 |   | IfcSpatialElement
IfcRelCoversBldgElements | RelatingBuildingElement | 1..1 |   | IfcElement
IfcRelCoversBldgElements | RelatedCoverings | 1..* | n | IfcCovering
IfcRelCoversSpaces | RelatingSpace | 1..1 |   | IfcSpace
IfcRelCoversSpaces | RelatedCoverings | 1..* | n | IfcCovering
IfcRelFillsElement | RelatingOpeningElement | 1..1 |   | IfcOpeningElement
IfcRelFillsElement | RelatedBuildingElement | 1..1 |   | IfcElement
IfcRelFlowControlElements | RelatedControlElements | 1..* | n | IfcDistributionControlElement
IfcRelFlowControlElements | RelatingFlowElement | 1..1 |   | IfcDistributionFlowElement
IfcRelInterferesElements | RelatingElement | 1..1 |   | IfcElement
IfcRelInterferesElements | RelatedElement | 1..1 |   | IfcElement
IfcRelReferencedInSpatialStructure | RelatedElements | 1..* | n | IfcProduct
IfcRelReferencedInSpatialStructure | RelatingStructure | 1..1 |   | IfcSpatialElement
IfcRelSequence | RelatingProcess | 1..1 |   | IfcProcess
IfcRelSequence | RelatedProcess | 1..1 |   | IfcProcess
IfcRelServicesBuildings | RelatingSystem | 1..1 |   | IfcSystem
IfcRelServicesBuildings | RelatedBuildings | 1..* | n | IfcSpatialElement
IfcRelSpaceBoundary | RelatingSpace | 1..1 |   | IfcExternalSpatialElement,IfcSpace
IfcRelSpaceBoundary | RelatedBuildingElement | 1..1 |   | IfcElement
IfcRelDeclares | RelatingContext | 1..1 |   | IfcContext
IfcRelDeclares | RelatedDefinitions | 1..* | n | IfcObjectDefinition,IfcPropertyDefinition
IfcRelAggregates | RelatingObject | 1..1 |   | IfcObjectDefinition
IfcRelAggregates | RelatedObjects | 1..* | n | IfcObjectDefinition
IfcRelNests | RelatingObject | 1..1 |   | IfcObjectDefinition
IfcRelNests | RelatedObjects | 1..* | y | IfcObjectDefinition
IfcRelProjectsElement | RelatingElement | 1..1 |   | IfcElement
IfcRelProjectsElement | RelatedFeatureElement | 1..1 |   | IfcFeatureElementAddition
IfcRelVoidsElement | RelatingBuildingElement | 1..1 |   | IfcElement
IfcRelVoidsElement | RelatedOpeningElement | 1..1 |   | IfcFeatureElementSubtraction
IfcRelDefinesByObject | RelatedObjects | 1..* | n | IfcObject
IfcRelDefinesByObject | RelatingObject | 1..1 |   | IfcObject
IfcRelDefinesByProperties | RelatedObjects | 1..* | n | IfcObjectDefinition
IfcRelDefinesByProperties | RelatingPropertyDefinition | 1..1 |   | IfcPropertySetDefinition,IfcPropertySetDefinitionSet
IfcRelDefinesByTemplate | RelatedPropertySets | 1..* | n | IfcPropertySetDefinition
IfcRelDefinesByTemplate | RelatingTemplate | 1..1 |   | IfcPropertySetTemplate
IfcRelDefinesByType | RelatedObjects | 1..* | n | IfcObject
IfcRelDefinesByType | RelatingType | 1..1 |   | IfcTypeObject

Reply:
Fair point. Tried an approach with the lookup table, as it may prove to be the approach that can be reused the most.

I think it would be neat, to move the dicts I created to a separate python .py file, not to clutter steps.py. 
Would you be ok with it? Some time ago, I recall, there was an agreement to try to avoid it.

Reply:
Agreed to use a .csv file. Implemented.